### PR TITLE
change: Enable load_run without name args in Transform env

### PR DESCRIPTION
**Pull request by qidewenwhen**
*Description of changes:* 
- As transform job team has helped to add a env var to supply the job source arn, it's a good time to enable load_run without name args in Transform env
- Clean up some TODO items in the unit/integ tests
- Clean up two unused clases

*Testing done:* unit tests and integ tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by claytonparnell — COMMENTED** [review]
/bot run slow-tests

**Comment by claytonparnell** [comment]
@qidewenwhen - sorry this has been open so long; could you please resolve merge conflict then we can get it merged?

**Comment by qidewenwhen** [comment]
> @qidewenwhen - sorry this has been open so long; could you please resolve merge conflict then we can get it merged?

Thanks @claytonparnell! Resolving the conflicts right now. 
This PR is still pending internal team review. Will bump it up again in internal team channel next Monday. 
Once we get internal approvals, we can move forward to merge.

File: src/sagemaker/experiments/_environment.py
Comment (danabens) [inline src/sagemaker/experiments/_environment.py:59]: i dont understand why these are args in the first place - they will never change and so should be constants

**Review by danabens — APPROVED** [review]

**Comment by danabens** [comment]
we need to talk to the client frameworks team about slow tests, just causing constant failures in PRs - not providing any value

```
(11 durations < 0.005s hidden.  Use -vv to show these durations.)
=========================== short test summary info ============================
FAILED tests/integ/test_inference_pipeline.py::test_inference_pipeline_model_deploy_and_update_endpoint
============= 1 failed, 15 passed, 1 warning in 3533.54s (0:58:53) =============
ERROR: InvocationError for command /codebuild/output/src083470416/src/github.com/aws/sagemaker-python-sdk/.tox/py310/bin/pytest --cov=sagemaker --cov-append tests/integ -m slow_test -n 16 --durations 0 (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py310: commands failed
```

File: src/sagemaker/experiments/_environment.py
Comment (qidewenwhen) [reply src/sagemaker/experiments/_environment.py:59]: This was kept when migrating the Experiment SDK to main SDK: https://github.com/aws/sagemaker-experiments/blob/caf7b147ddc6fe44f40b07ca3a5e011d9c57cfae/src/smexperiments/_environment.py#L47.
In Exp SDK, the `TrialComponentEnvironment.load` is a public facing method, I guess leaving these local vars may be useful when calling/testing it in different env.
However, as given it's internal method in our main SDK use case, agree that these local args are not necessary. Will remove them

Comment (qidewenwhen) [reply src/sagemaker/experiments/_environment.py:59]: This [unit test](https://github.com/aws/sagemaker-python-sdk/blob/a94a3b17fb7f3aa46906869cba0a24cb5522d4d2/tests/unit/sagemaker/experiments/test_environment.py#L58) migrated from Exp SDK seems prove my assumption.
With these args, they make this method to be more flexible and much easier to test in terms of mocking the processing job env (and the transform job env in the future as well once the transform team gets the config file ready for us).
Thus, after a second thought, I'll keep this behavior.

**Review by leoroy — APPROVED** [review]

**Review by claytonparnell — APPROVED** [review]

**Pull request merged**
